pkgs/applications/science/chemistry/pymol/default.nix

```
@@ -18,35 +13,44 @@ let
     desktopName = "PyMol Molecular Graphics System";
     genericName = "Molecular Modeler";
     comment = description;
+    icon = pname;
     mimeType = "chemical/x-pdb;chemical/x-mdl-molfile;chemical/x-mol2;chemical/seq-aa-fasta;chemical/seq-na-fasta;chemical/x-xyz;chemical/x-mdl-sdf;";
     categories = "Graphics;Education;Science;Chemistry;";
   };
 in
-python3Packages.buildPythonApplication {
-  name = "pymol-${version}";
-  src = fetchurl {
-    url = "mirror://sourceforge/project/pymol/pymol/${ver_maj}/pymol-v${version}.tar.bz2";
-    sha256 = "1qpacd5w4r9a0nm5iqmkd92ym3ai00dp7v61cwd6jgakk6wfps3s";
+python3Packages.buildPythonApplication rec {
+  inherit pname;
+  version = "2.3.0";
+  src = fetchFromGitHub {
+    owner = "schrodinger";
+    repo = "pymol-open-source";
+    rev = "v${version}";
+    sha256 = "175cqi6gfmvv49i3ws19254m7ljs53fy6y82fm1ywshq2h2c93jh";
   };
 
-  buildInputs = [ python3Packages.numpy glew freeglut libpng libxml2 tk freetype msgpack ];
-  NIX_CFLAGS_COMPILE = "-I ${libxml2.dev}/include/libxml2";
+  buildInputs = [ python3Packages.numpy glew glm freeglut libpng libxml2 tk freetype msgpack ];
+  NIX_CFLAGS_COMPILE = "-I ${libxml2.dev}/include/libxml2 -Wno-error=format-security";
+
+  setupPyBuildFlags = [ "--glut" ];
 
   installPhase = ''
-    python setup.py install --home=$out
-    cp -r ${desktopItem}/share/ $out/
+    python setup.py install --home="$out"
     runHook postInstall
   '';
 
   postInstall = with python3Packages; ''
     wrapProgram $out/bin/pymol \
-      --prefix PYTHONPATH : ${makeSearchPathOutput "lib" python3.sitePackages [ Pmw tkinter ]}
+      --prefix PYTHONPATH : ${lib.makeSearchPathOutput "lib" python3.sitePackages [ Pmw tkinter ]}
+
+    mkdir -p "$out/share/icons/"
+    ln -s ../../lib/python/pymol/pymol_path/data/pymol/icons/icon2.svg "$out/share/icons/pymol.svg"
+    cp -r "${desktopItem}/share/applications/" "$out/share/"
   '';
 
   meta = {
     description = description;
     homepage = https://www.pymol.org/;
-    license = licenses.psfl;
-    broken = true;
+    license = lib.licenses.psfl;
```

```suggestion
    license = licenses.psfl;
```
Also the license seems to be outdated. It should be some king of BSD license: https://github.com/schrodinger/pymol-open-source#license

---

Oops, thanks for checking that. It seems to be a slightly modified MIT license. They add more explicitly that you can distribute modified copies, and clarify that the license only applies to built-in documentation rather than any other documentation Schroedinger may provide/sell.